components/butterfly/src/rumor.rs

```diff
@@ -55,6 +58,58 @@ lazy_static! {
                                   &["rumor"]).unwrap();
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Expiration(Option<DateTime<Utc>>);
+
+impl fmt::Display for Expiration {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let d = match self.0 {
+            Some(e) => e.to_rfc3339(),
+            None => String::from("None"),
+        };
+
+        write!(f, "{}", d)
+    }
+}
+
+impl Expiration {
+    // This is more of a generic expiration date that we can apply whenever we have a rumor we
+    // don't need to keep around any more.
+    pub fn soon() -> Self { Self(Some(Self::soon_date())) }
+
+    pub fn never() -> Self { Self(None) }
```

Comment:
This can be `pub(crate)`.

Reply:
I notice that we're using `never` as a default value... perhaps it's worth encoding that explicitly in a `Default` implementation, and then documenting _why_ that's a useful default to have.

Reply:
I will change it to `pub(crate)`.  I had initially implemented `Default` for this, but one of @baumanj 's suggestions was to remove `Default` and explicitly use `never()` whenever we create a value, so that's how we ended up here.

Reply:
I remember @baumanj's suggestion, and agree that it wasn't immediately obvious that `never()` was the "right" default.

However, judging from its usage in the rest of this PR, it appears to be the de facto default value. Given that, I felt it might be good to at least formally state that, rather than having it just be an informal undocumented convention.